corehq/apps/hqcase/tests/test_case_helper.py

```diff
@@ -293,6 +294,26 @@ def test_get_user_duck_username(self):
             self.assertEqual(user.username, commcare_user.username)
 
 
+@generate_cases([(f.name,) for f in CommCareCase._meta.fields], cls=TestCase)
```

Comment:
Is `invalid_fields` a subset of `CommCareCase._meta.fields`? Are there any invalid fields that are not in `CommCareCase._meta.fields`? Things like

https://github.com/dimagi/commcare-hq/blob/41773d5ec28a9e05189f493d5d35be95607fac63/corehq/apps/case_search/const.py#L33-L38

Reply:
Anything that isn't in [`BaseJsonCaseChange._properties_by_key`](https://github.com/dimagi/commcare-hq/blob/834ce5325108373b03eed3a2b46bdfb429affe08/corehq/apps/hqcase/api/updates.py#L83) is an invalid field.

:thinking: I'm doing this the wrong way round, aren't I?

...

fae79da2d2a